app/component/map/map-with-tracking.cjsx

```diff
@@ -0,0 +1,98 @@
+React       = require 'react'
+timeUtils   = require '../../util/time-utils'
+cx          = require 'classnames'
+ComponentUsageExample = require '../documentation/component-usage-example'
+Example               = require '../documentation/example-data'
+Map                   = require './map.cjsx'
+ToggleMapTracking     = require '../navigation/toggle-map-tracking'
+MapTrackActions = require '../../action/map-track-actions'
+
+class MapWithTracking extends React.Component
+  @contextTypes:
+    getStore: React.PropTypes.func.isRequired
+    executeAction: React.PropTypes.func.isRequired
+
+  constructor: ->
+    super
+    #Check if we have a position already
+    locationState = @context.getStore('PositionStore').getLocationState()
+    if locationState.hasLocation
+      initialZoom = 16
+      initialLat = locationState.lat
+      initialLon = locationState.lon
+    else
+      initialZoom = 11
+      initialLat = 60.17332
+      initialLon = 24.94102
+
```

Comment:
Good point. Actually most of the above which is in map.cjsx is not needed there. It only checks whether  it has a position or not, and uses that information to insert a position marker. I can delete the parts which are not needed in map.cjsx. Also I too agree with the config settings for those. I'll put them as variables in there.